thunder/dynamo/benchmark_utils.py

```diff
@@ -224,6 +224,34 @@ def cleanup(self):
             pass
 
 
+class TimerWithCUDAMemoryUsage:
```

Comment:
Cool! Great modularity that this is a separate timer. Even if the same fx graph shows a speed issue and a memory issue, it's probably simplest to create two scripts, one for each issue, and separate bugs.